### DISCUSSION TITLE
Flowise Integration

### DISCUSSION THREAD

**Discussion by mynumu** [danny-avila/LibreChat#1765] [General]
Hey new to librechat..amazing work! I see you can write your own langchain tools/plugins…I can do that but just wondering if there’s a way to drop in the code from Flowise as I use this for a lot of langchain stuff! FW has an API endpoint…I’m a very low coder so not sure what exactly is needed. Flowise is a JS GUI for langchain thanks

**Comment by danny-avila** [comment]
Would love to integrate Flowise, was discussed here: https://discord.com/channels/1086345563026489514/1105648303909126144/1198402537733230704

I'm going to add this to the roadmap

**Reply by Daniel-DDV** [reply]
That would be amazing @danny-avila

**Reply by taowang1993** [reply]
Is flowise or langflow or dify integrated already? I cannot see them on the UI. 

Do I need to build a plugin/tool to integrate them?

**Reply by jars08** [reply]
I'd love to see the integration of a service like flowise, langflow or dify. I wasn't able to configure any of them as a custom endpoint yet.

**Reply by taowang1993** [reply]
> I'd love to see the integration of a service like flowise, langflow or dify. I wasn't able to configure any of them as a custom endpoint yet.

You can use this https://github.com/open-webui/pipelines

Connect pipeline as custom endpoint with librechat. But you need to build the python pipeline by yourself. 

I only succeeded with flowise so far. I tried to connect Dify today, but did not have any luck.

**Reply by danny-avila** [reply]
There's no need to use a "pipeline" for this, as native support is incoming

**Reply by taowang1993** [reply]
> There's no need to use a "pipeline" for this, as native support is incoming

OMG!!! I can't wait! Thank you so much for your work!! You are awesome!

**Reply by Daniel-DDV** [reply]
Really looking forward to this. This is going to be an extremely powerful addition to LibreChat. Any sight on when @danny-avila ?

**Reply by HenryHengZJ** [reply]
🚀🚀 lets go

**Reply by danny-avila** [reply]
I would like to start this month.. with AI, there is always something exciting, and my current work on a native-to-librechat agentic system has been taking a lot of my interest/time. Realistically, it makes sense for me to work on an integration of someone else's project after working on our own, I hope this is understandable.

**Reply by jdmasa** [reply]
Hi all. First thanks @danny-avila for making LibreChat available for the community.
I would like to know if the Flowise integration was finally done or should I look for a workaround?
Thanks and regards
